.github/workflows/ci.yml

```diff
@@ -0,0 +1,127 @@
+name: ci
+
+on: [push, pull_request]
+
+env:
+  NODE_VERSION: 10
+  SOLR_VERSION: 6.6.5
+
+jobs:
+
+  build-static:
+    name: Build static files
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+    - name: Cache node modules
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: npm-${{ hashFiles('package-lock.json') }}
+        restore-keys: |
+          npm-${{ hashFiles('package-lock.json') }}
+          npm-
+    - name: Install dependencies
+      run: npm ci
+    - name: Run webpack build
+      run: npm run build:prod
+    - name: Upload built files
+      uses: actions/upload-artifact@v2
+      with:
+        name: bundles
+        path: bundles/
+    - name: Upload stats file
+      uses: actions/upload-artifact@v2
+      with:
+        name: webpack-stats.json
+        path: webpack-stats.json
+
+  ts-unit:
+    name: TypeScript unit tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+    - name: Cache node modules
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: npm-${{ hashFiles('package-lock.json') }}
+        restore-keys: |
+          npm-${{ hashFiles('package-lock.json') }}
+          npm-
+    - name: Install dependencies
+      run: npm ci
+    - name: Run Jest
+      run: npx jest --collectCoverage
+    - name: Upload test coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        flags: typescript
+
+  py-unit:
+    name: Python unit tests
+    runs-on: ubuntu-latest
+    needs: build-static
+    strategy:
+      matrix:
+        python: [3.5, 3.6, 3.7]
+    services:
+      db:
+        image: mariadb
+        env:
+          MYSQL_DATABASE: sandco
+          MYSQL_ROOT_PASSWORD: sandco
+        ports:
+        - 3306:3306
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Cache pip
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ matrix.python }}-${{ hashFiles('requirements.txt') }}
+        restore-keys: |
+          pip-${{ matrix.python }}-${{ hashFiles('requirements.txt') }}
+          pip-${{ matrix.python }}
+          pip-
+    - name: Install dependencies
+      run: |
+        pip install wheel
```

Comment:
Should we refactor requirements files using the directory structure with includes so that you can use `requirements/dev.txt` (or `requirements/test.txt`) and only one pip install command is needed?

I don't know why `wheel` is installed separately, but we should probably add it to dev requirements.

Reply:
makes sense to me. I was getting build warnings that it was using non-wheel versions of packages (?) because wheel wasn't installed, but I didn't know if it was worth adding to dev-requirements, so I just added it as its own step here.